datamade/just-spaces
# Add API endpoint to return ACS data for DAD charts

## Overview

Connects #100.

Update data imports for ACS data and add an endpoint that will return appropriate ACS data for a given survey question in the DAD.

## Testing Instructions

### Tests and migrations

- Confirm that new tests pass on Travis
- Run migrations
- Run the new `import_data` command and confirm that it loads data:

```
python manage.py import_data
```

### Display of census geography widget and retrieval of ACS data

- Navigate to a survey submitted detail that has a question that is ACS-elegible (check `fobi_custom.plugins.form_elements.fields.types.TYPES_TO_ACS_VARIABLES` for a list of valid question types)
- Create a new chart for the eligible question and confirm that it displays a form field for Census geographies
- Open your browser's dev console, select a geographic level, and confirm that it prints ACS data to the console

### Proper hiding of census geography widget

- On the same page, change the `Primary data source` to the null value (`-----`) and confirm that the page hides the Census geographies field
- Navigate to a survey submitted detail that has a question that is **not** ACS-eligible
- Create a new chart for the ineligible question and confirm that the page does not show a Census geographies field
